staging/src/k8s.io/apimachinery/pkg/util/strategicpatch/patch.go

```diff
@@ -1842,37 +1845,52 @@ func mergingMapFieldsHaveConflicts(
 			return false, nil
 		}
 		// Check the individual keys.
-		return mapsHaveConflicts(leftType, rightType, fieldType)
+		return mapsHaveConflicts(leftType, rightType, schema)
 
 	case []interface{}:
 		rightType, ok := right.([]interface{})
 		if !ok {
 			return true, nil
 		}
-		return slicesHaveConflicts(leftType, rightType, fieldType, fieldPatchStrategy, fieldPatchMergeKey)
-
+		return slicesHaveConflicts(leftType, rightType, schema, fieldPatchStrategy, fieldPatchMergeKey)
 	case string, float64, bool, int, int64, nil:
 		return !reflect.DeepEqual(left, right), nil
 	default:
 		return true, fmt.Errorf("unknown type: %v", reflect.TypeOf(left))
 	}
 }
 
-func mapsHaveConflicts(typedLeft, typedRight map[string]interface{}, structType reflect.Type) (bool, error) {
+func mapsHaveConflicts(typedLeft, typedRight map[string]interface{}, schema LookupPatchMeta) (bool, error) {
 	for key, leftValue := range typedLeft {
 		if key != directiveMarker && key != retainKeysDirective {
 			if rightValue, ok := typedRight[key]; ok {
-				fieldType, fieldPatchStrategies, fieldPatchMergeKey, err := forkedjson.LookupPatchMetadata(structType, key)
-				if err != nil {
-					return true, err
-				}
-				_, patchStrategy, err := extractRetainKeysPatchStrategy(fieldPatchStrategies)
-				if err != nil {
-					return true, err
+				var subschema LookupPatchMeta
+				var patchMeta PatchMeta
+				var patchStrategy string
+				var err error
+				switch leftValue.(type) {
+				case []interface{}:
+					subschema, patchMeta, err = schema.LookupPatchMetadataForSlice(key)
+					if err != nil {
+						return true, err
+					}
+					_, patchStrategy, err = extractRetainKeysPatchStrategy(patchMeta.patchStrategies)
+					if err != nil {
+						return true, err
+					}
+				case map[string]interface{}:
```

Comment:
this applies to all non-slice values, not just map child fields, right?

Reply:
No. This applies only to non-primitive maps.

For primitive maps, we do nothing. E.g. we should not lookup patch metadata from the struct of `map[string]int`.
So nil or empty subschema, patchStrategy, mergeKey will be pass to `mergingMapFieldsHaveConflicts()`.